api/caching/tasks.py

```diff
@@ -131,11 +132,62 @@ def update_storage_usage_cache(target_id, target_guid, per_page=500000):
             offset += count
 
     key = cache_settings.STORAGE_USAGE_KEY.format(target_id=target_guid)
-    storage_usage_cache.set(key, storage_usage_total, cache_settings.FIVE_MIN_TIMEOUT)
+    storage_usage_cache.set(key, storage_usage_total, cache_settings.ONE_DAY_TIMEOUT)
 
 
 def update_storage_usage(target):
     Preprint = apps.get_model('osf.preprint')
 
     if settings.ENABLE_STORAGE_USAGE_CACHE and not isinstance(target, Preprint) and not target.is_quickfiles:
         enqueue_postcommit_task(update_storage_usage_cache, (target.id, target._id,), {}, celery=True)
+
+def update_storage_usage_with_size(payload):
+    BaseFileNode = apps.get_model('osf.basefilenode')
+    AbstractNode = apps.get_model('osf.abstractnode')
+    Preprint = apps.get_model('osf.preprint')
+
+    metadata = payload.get('metadata') or payload.get('destination')
+
+    if not metadata.get('nid'):
+        return
+    target_node = AbstractNode.load(metadata['nid'])
+
+    if isinstance(target_node, Preprint) or target_node.is_quickfiles:
+        return
+
+    action = payload['action']
+    target_file_id = metadata['path'].replace('/', '')
+    size = metadata.get('size', 0)
+
+    current_usage = target_node.storage_usage or 0
+    target_file = BaseFileNode.load(target_file_id)
+
+    if action in ['create', 'update']:
+        current_usage += size
+    elif action == 'delete':
+        size = target_file.versions.aggregate(Sum('size'))['size__sum']
+        if current_usage < size:
+            current_usage = 0
+        else:
+            current_usage -= target_file.versions.aggregate(Sum('size'))['size__sum']
+    elif action == 'move':
```

Comment:
Note from pairing: consider supporting "copy" here and only run the subtraction if the action is "move"